packages/bsky/src/notifications.ts

```diff
@@ -0,0 +1,253 @@
+          notifsToSend.push({
+            tokens: [token],
+            platform: 1,
+            title: title,
+            message: body,
+            topic: appId,
+            data: {
+              reason: notif.reason,
+              recordUri: notif.recordUri,
+              recordCid: notif.recordCid,
+            },
+          })
+        } else if (platform === 'android') {
+          notifsToSend.push({
+            tokens: [token],
+            platform: 2,
+            title: title,
+            message: body,
+            topic: appId,
+            data: {
+              reason: notif.reason,
+              recordUri: notif.recordUri,
+              recordCid: notif.recordCid,
+            },
+          })
+        } else {
+          // TODO: Handle web notifs
+        }
+      }
+    }
+    return notifsToSend
+  }
+
+  /**  1. Get the user's token (APNS or FCM for iOS and Android respectively) from the database
+    User token will be in the format:
+        did || token || platform (1 = iOS, 2 = Android, 3 = Web)
+    2. Send notification to `gorush` server with token
+    Notification will be in the format:
+    "notifications": [
+      {
+        "tokens": string[],
+        "platform": 1 | 2,
+        "message": string,
+        "title": string,
+        "priority": "normal" | "high",
+        "image": string, (Android only)
+        "expiration": number, (iOS only)
+        "badge": number, (iOS only)
+      }
+    ]
+    3. `gorush` will send notification to APNS or FCM
+    4.  store response from `gorush` which contains the ID of the notification
+    5. If notification needs to be updated or deleted, find the ID of the notification from the database and send a new notification to `gorush` with the ID (repeat step 2)
+  */
+  async sendPushNotifications(notifications: PushNotification[]) {
+    // if no notifications, skip and return early
+    if (!notifications || notifications.length === 0) {
+      return
+    }
+    // if pushEndpoint is not defined, we are not running in the indexer service, so we can't send push notifications
+    if (!this.pushEndpoint) {
+      throw new Error('Push endpoint not defined')
+    }
+    await axios.post(
+      this.pushEndpoint,
+      {
+        notifications: notifications,
+      },
+      {
+        headers: {
+          'Content-Type': 'application/json',
+          accept: 'application/json',
+        },
+      },
+    )
+  }
+
+  async registerDeviceForPushNotifications(
+    did: string,
+    platform: Platform,
+    token: string,
+    appId: string,
+    endpoint: string,
+  ) {
+    // if token doesn't exist, insert it, on conflict do nothing
+    await this.db.db
+      .insertInto('notification_push_token')
+      .values({
+        did,
+        token,
+        platform,
+        appId,
+        endpoint,
+      })
+      .onConflict((oc) => oc.doNothing())
+      .execute()
+  }
+
+  async getNotificationDisplayAttributes(notif: InsertableNotif) {
```

Comment:
a return type on this one might be helpful to clarify that it can return `undefined`